### PR TITLE
feat(signup): allow DRDD to sign up

### PR DESCRIPTION
**Pull request by kwiss**
- [x] add direction type
- [x] add more validation
- [x] test all signup type (we never know)
- [x] fix the tests
- [x] add api version choice form signup
- [x] add choice for ti

### CODE REVIEW

File: packages/app/src/components/communComponents/Api.js [hunk 1]
```diff
@@ -7,23 +7,11 @@ const {
 } = getConfig();
 
 // forceLogin: redirect user to /login when receiving a 401
-const apiFetch = (route, params, options = { forceLogin: true }) => {
+const apiFetch = (route, params, options = { forceLogin: true, hasUrl: false }) => {
```
Comment (kwiss) [inline packages/app/src/components/communComponents/Api.js]: replace with api version

File: packages/api/db/errors.js [hunk 2]
```diff
@@ -9,106 +9,185 @@ const {
   DataError
 } = require("objection-db-errors");
 
+const getErrorMessage = err => {
+  switch (err.constraint) {
+    case "users_email_unique":
+      return "Cet email est déjà utilisé.";
```
Comment (douglasduteil) [inline packages/api/db/errors.js]: ... We should ask ourself be french on the API...

File: packages/app/src/components/inscription/Form.js [hunk 3]
```diff
@@ -176,19 +172,49 @@ class Form extends React.Component {
...
+                  />
                 </tr>
               </tbody>
             </table>
+            {!isAgent && userType && (
+              <Resolve
+                promises={[() => getTis()]}
+                render={({ status, result }) => (
+                  <div style={{ margin: "20px 0" }}>
+                    <div style={{ fontSize: "1.2em", fontWeight: "bold", margin: "20px 0" }}>
```
Comment (douglasduteil) [inline packages/app/src/components/inscription/Form.js]: Why inline style and not styled-components ?

File: packages/app/src/components/inscription/Form.js [hunk 4]
```diff
@@ -176,19 +172,49 @@ class Form extends React.Component {
...
+                      Choisissez les tribunaux sur lesquels vous exercez :
+                    </div>
+                    {status === "success" && <TiSelector onChange={this.setTis} tis={result[0]} />}
```
Comment (douglasduteil) [inline packages/app/src/components/inscription/Form.js]: Looks like a switch to me :smirk:

File: packages/app/src/components/inscription/Form.js [hunk 5]
```diff
@@ -176,19 +172,49 @@ class Form extends React.Component {
...
+
             {this.state.status === "error" && (
-              <div style={{ textAlign: "center", color: "red", fontSize: "1.1em" }}>
-                {this.state.message}
+              <div>
+                {this.state.message.map(error => {
+                  return (
+                    <div
+                      key={error.msg}
+                      style={{ textAlign: "center", color: "red", fontSize: "1.1em" }}
```
Comment (douglasduteil) [inline packages/app/src/components/inscription/Form.js]: Why inline style and not styled-components ?

File: packages/app/src/components/inscription/uiSchema.json [hunk 6]
```diff
@@ -74,5 +76,9 @@
   "cabinet": {
     "ui:title": "Cabinet",
     "ui:placeholder": "cabinet"
+  },
+  "type": {
+    "ui:title": "Type d'agent",
+    "ui:placeholder": "Type d'agent"
   }
-}
+}
```
Comment (douglasduteil) [inline packages/app/src/components/inscription/uiSchema.json]: missing EOL

File: docker-compose.yaml [hunk 7]
```diff
@@ -37,11 +37,13 @@ services:
...
     depends_on:
       - db
+      
+    restart: always
     environment:
       HASURA_GRAPHQL_DATABASE_URL: postgres://postgres:test@db:5432/emjpm_dev
       HASURA_GRAPHQL_ENABLE_CONSOLE: "true" # set to "false" to disable console
       HASURA_GRAPHQL_ADMIN_SECRET: secret
-      HASURA_GRAPHQL_JWT_SECRET: '{"type": "RS256","jwk_url": "$API_URL_DOCKER:4000/auth/v2/jwks"}'
+      HASURA_GRAPHQL_JWT_SECRET: '{"type": "RS256","jwk_url": "$API_URL_DOCKER:4000/api/v2/auth/jwks"}'
```
[the remark this answers is not in the document]
Comment (kwiss) [reply docker-compose.yaml]: yep

File: packages/api/__test__/e2e/inscription/POST_inscription.test.js [hunk 8]
```diff
@@ -71,13 +74,13 @@ const defaultRegister = {
...
   const response = await request(server)
-    .post("/api/v1/inscription/mandataires")
+    .post("/api/v2/auth/signup")
     .send(defaultRegister);
   expect(response.body).toMatchInlineSnapshot(`
-    Object {
-      "success": true,
-    }
-  `);
+            Object {
```
[the remark this answers is not in the document]
Comment (kwiss) [reply packages/api/__test__/e2e/inscription/POST_inscription.test.js]: i'll check if it's a jest problem when generating inline snapshop

File: packages/app/src/components/inscription/Form.js (hunk 4 above)
Comment (kwiss) [reply packages/app/src/components/inscription/Form.js]: we should create an issue for that legacy

File: packages/app/src/components/inscription/uiSchema.json (hunk 6 above)
Comment (kwiss) [reply packages/app/src/components/inscription/uiSchema.json]: we should create an issue for that legacy

File: packages/app/src/components/inscription/Form.js (hunk 5 above)
Comment (kwiss) [reply packages/app/src/components/inscription/Form.js]: we should create an issue for that legacy

**Pull request merged**